api/utils/countly-request/index.js

```diff
@@ -150,39 +150,83 @@ var convertOptionsToGot = function(options) {
+                    params.callback(error);
+                });
+        }
+        else {
+            got(params.uri, params.options)
+                .then(response => {
+                    params.callback(null, response, response.body);
+                })
+                .catch(error => {
+                    params.callback(error);
+                });
+        }
     }
 
+    // eslint-disable-next-line require-jsdoc
+    function post(uri, options, callback, config) {
+        var params = initParams(uri, options, callback, config);
+        if (params.options && (params.options.url || params.options.uri)) {
+            if (params.options.form && params.options.form.fileStream && params.options.form.fileField) {
+                // If options include a form, use uploadFormFile
+                const { url, form } = params.options;
+                uploadFormFile(url || params.options.uri, form, params.callback);
+            }
+            else {
+                // Make the request using got
+                got.post(params.options)
+                    .then(response => {
+                        // Call the callback with the response data
+                        params.callback(null, response, response.body);
+                    })
+                    .catch(error => {
+                        // Call the callback with the error
+                        params.callback(error);
+                    });
+            }
+        }
+        else {
+            // Make the request using got
+            got.post(params.uri, params.options)
+                .then(response => {
+                    params.callback(null, response, response.body);
+                })
+                .catch(error => {
+                    // Call the callback with the error
+                    params.callback(error);
+                });
+        }
+    }
 
-    const params = initParams(uri, options, callback);
 
-    if (params.options && (params.options.url || params.options.uri)) {
-        // Make the request using got
-        got(params.options)
-            .then(response => {
-                // Call the callback with the response data
-                params.callback(null, response, response.body);
-            })
-            .catch(error => {
-                // Call the callback with the error
-                params.callback(error);
-            });
-    }
-    else {
-        // Make the request using got
-        got(params.uri, params.options)
-            .then(response => {
-                params.callback(null, response, response.body);
-            })
-            .catch(error => {
-                // Call the callback with the error
-                params.callback(error);
-            });
+    function get(uri, options, callback, config) {
+        module.exports(uri, options, callback, config);
     }
 
+    requestFunction.post = post;
+    requestFunction.get = get;
+
+    return requestFunction;
```

Comment:
We probably should move `convertOptionsToGot` here

Reply:
I'have added to convertOptionsToGot to exports.